### ISSUE TITLE
GstGL GLMixer processTextures function is generated as invalid function

### ISSUE THREAD

**Issue by BwackNinja** [jwharm/java-gi#175]
This seems to be a corner-case in code generation.

The virtual-method and method signatures for `processTextures` in `GLMixer` don't match. One expects a `GstGLMemory out_tex`, and the other expects `GstBuffer out_buf`. This results in generated code using `outTex` without declaring it.

Despite having the same name, `processTextures(Buffer outbuf)` and the virtual method `processTextures(GLMemory outtex)` aren't the same method. The one taking the `Buffer` should call `gst_gl_mixer_process_textures`, while the one taking the `GLMemory` is called within `gst_gl_mixer_process_textures` and is implemented/overriden by subclasses.

Looking through gstreamer git, this was introduced 12 years ago: https://gitlab.freedesktop.org/gstreamer/gstreamer/-/commit/89569276c563d3343417ea5e34bddcd0714131d7

**Comment by jwharm** [comment]
Thanks for the issue reports!

This one is a bug in GStreamer. You can work around it by editing the gir file: simply remove the `invoker="process_textures"` attribute from the `<virtual-method>` element.

I can't fix this in java-gi, unless I start second-guessing the `invoker` attribute, but I'd rather not.

**Comment by BwackNinja** [comment]
Thanks for addressing these issues so quickly!

If this is a problem in the gir file and you have to patch similar issues even in the GObject and Gio gir files, should this be an issue opened upstream? I couldn't find a related issue in gobject-introspection or a reference to similar issues in other bindings. The rust bindings don't seem to parse the `invoker` attribute at all, which I believe is what limits subclassing support as referenced here: https://github.com/gtk-rs/gir/pull/1463

Additionally or alternatively, checking whether the parameters match in `VirtualMethod.invoker()` and `Method.invokerFor()` should be enough for this as well as removing the corresponding patches for GObject and Gio.

**Comment by jwharm** [comment]
Agreed, checking if the parameters match is a reasonable check. I’ll try it out.

**Comment by jwharm** [comment]
The issue with GLMixer is now resolved, when the bindings are generated with the newest java-gi from the main branch.
I'm planning a new java-gi release soon.
